programs/ssl/ssl_base64_dump.c

```diff
@@ -360,17 +359,54 @@ void print_deserialized_ssl_session( const uint8_t *ssl, size_t len,
         }
         else
         {
-            peer_cert_digest_type = (mbedtls_md_type_t) *ssl++;
-            printf( "\tpeer_cert_digest_type: %d\n", (int)peer_cert_digest_type );
+            printf( "\tMessage-Digest type: " );
 
-            peer_cert_digest_len  = (size_t) *ssl++;
-            printf_dbg( "peer_cert_digest_len: %zu\n", peer_cert_digest_len );
+            mbedtls_md_type_t md_type = (mbedtls_md_type_t) *ssl++;
+            switch( md_type )
+            {
+                case MBEDTLS_MD_NONE:
+                    printf( "none\n" );
+                    break;
+                case MBEDTLS_MD_MD2:
+                    printf( "MD2\n" );
+                    break;
+                case MBEDTLS_MD_MD4:
+                    printf( "MD4\n" );
+                    break;
+                case MBEDTLS_MD_MD5:
+                    printf( "MD5\n" );
+                    break;
+                case MBEDTLS_MD_SHA1:
+                    printf( "SHA1\n" );
+                    break;
+                case MBEDTLS_MD_SHA224:
+                    printf( "SHA224\n" );
+                    break;
+                case MBEDTLS_MD_SHA256:
+                    printf( "SHA256\n" );
+                    break;
+                case MBEDTLS_MD_SHA384:
+                    printf( "SHA384\n" );
+                    break;
+                case MBEDTLS_MD_SHA512:
+                    printf( "SHA512\n" );
+                    break;
+                case MBEDTLS_MD_RIPEMD160:
+                    printf( "RIPEMD160\n" );
```

Comment:
Do we not have some cipher printing code in the library already that we could use?

Reply:
Yes, there is `mbedtls_md_info_from_type()`, but I decided to write my own due to the fact that `mbedtls_md_info_from_type()` will not work for algorithms with are turned off in the mbedTLS configuration